### ISSUE TITLE
currency format

### ISSUE THREAD

**Issue by gour** [sdementen/piecash#119]
Hello,

I'd like to migrate from GnuCash to (h)ledger and using 'ledger' exports runs fine, but wonder if it is possible to convert data taking into consideration that e.g. for my native currency (Croatian kuna) and EUR the commodity format is:

```
commodity HRK
  format HRK 9.999.999,00

commodity EUR
  format EUR 9.999.999,00
```
while for USD it is:

```
commodity USD
  format USD 9,999,999.00
```

At least, I'd Iike to convert to `1.000.000,00' format since I only have few transaction where USD is involved?

**Edited:** Fixed separators in the code example.

**Comment by sdementen** [comment]
Hello, I do not understand clearly the issue...
Could you describe what you are getting now and what you would expect instead?

**Comment by gour** [comment]
When I convert my data to (h)ledger format, Piecash converts everything into e.g. `123.00` format for my transactions involving HRK & EUR currencies, while official format for those currencies is to use decimal comma instead of decimal point - e.g. `123,00`.

The above mentioned commodity declarations as defined for [hledger](https://hledger.org/manual.html#declaring-commodities) shows how the amounts in transactions should be rendered, so I wonder if there is possibility in Piecash to define decimal comma (',') instead of decimal point ('.') as separator?

**Comment by sdementen** [comment]
Ok, all you formats where with . as decimal separator but you meant some where with , as deci00

**Comment by gour** [comment]
> Or do you want the ability to specify per currency the format? 

That would be nice!

> If so, what would be your preferred way of doing so?

Well, e.g. official format for Euro currency is to use decimal comma as decimal separator, e.g. `123,45` EUR instead of e.g. `123.45 USD`. If you check Gnucash, it honors rendering of amounts for each supported currency based on the `locale` settings, so I believe the best would be to have "format per commodity/currency" 

See [this](https://www.thefinancials.com/?SubSectionID=curformat) document.

**Comment by sdementen** [comment]
could you try the new branch ?

**Comment by sdementen** [comment]
@gour are you sure the ledger format allows something like

2019-06-01 FooDesc
	;BazNote
	Assets:Current Assets:Cash in Wallet      100,00 €
	Assets:Current Assets:Checking Account    -20,00 €
	Assets:Current Assets:Savings Account     -80,00 €

**Comment by gour** [comment]
> could you try the new branch ?

I've tried, but noticed two things:

1. the transaction still look like:

```2019-05-23 hrana (Metss, HL)
        Expenses:hrk:hrana                        HRK 36.00
        Assets:hrk:gotovina                       HRK -36.00
```

iow. still decimal point instead of decimal comma.

2. I see the script contains the following snippet:

```
[...]
try:
    from money import Money
except ImportError:
    Money = None
```

but the script did not complain although I did not have Money installed?

Lastly,

here is my locale output:

```
$ locale                                                      
LANG=hr_HR.UTF-8
LANGUAGE=en_GB:en
LC_CTYPE=hr_HR.UTF-8
LC_NUMERIC=hr_HR.UTF-8
LC_TIME=hr_HR.UTF-8
LC_COLLATE=hr_HR.UTF-8
LC_MONETARY=hr_HR.UTF-8
LC_MESSAGES=en_GB.UTF-8
LC_PAPER=hr_HR.UTF-8
LC_NAME=hr_HR.UTF-8
LC_ADDRESS=hr_HR.UTF-8
LC_TELEPHONE=hr_HR.UTF-8
LC_MEASUREMENT=hr_HR.UTF-8
LC_IDENTIFICATION=hr_HR.UTF-8
LC_ALL=
```

and `LC_MONETARY` is properly set to `hr_HR.UTF-8`

Any hint?

**Comment by gour** [comment]
> @gour are you sure the ledger format allows something like
> 
> 2019-06-01 FooDesc
> ;BazNote
> Assets:Current Assets:Cash in Wallet 100,00 €
> Assets:Current Assets:Checking Account -20,00 €
> Assets:Current Assets:Savings Account -80,00 €

Yes, here is my last entry in (h)ledger journal from today:

```
2019-06-10 hrana (Konzum, MS)
    Expenses:hrk:hrana                         38,79 HRK
    Assets:hrk:banka:me                      -38,79 HRK
```

**Comment by gour** [comment]
> could you try the new branch ?

It looks good! Thank you!!

One thing which I notice is that the price database is not converted, iow. I still get:

```
P 2013-02-01 00:00:00 EUR 7.51 HRK
```

**Comment by sdementen** [comment]
ok, I will fix also the prices.
what I meant is "does ledger accepts € instead of EUR as currency ?"
or should the currency still be EUR?

**Comment by gour** [comment]
> ok, I will fix also the prices.

I forgot to also mention txns which include prices like

```
2015-05-05 mjenjačnica (MS)
	Assets:hrk:gotovina                          2256.00 HRK @@ 300.00 EUR ;   hrk                 
	Assets:eur:gotovina                       -300,00 EUR ;   eur                 
```
> what I meant is "does ledger accepts € instead of EUR as currency ?"
> or should the currency still be EUR?

I just tried to replace 'USD' with '$' in my hledger and it works. Iirc, both hledger and ledger allows it, while Beancount is more strict in regards.

**Comment by sdementen** [comment]
when you declare the commodity, should you also use the symbol then
commodity EUR
	note Euro
==>
commodity €
	note Euro

and in the account definition, should the check commodity use the symbol too 
account Assets
	note Assets
	check commodity == "EUR"
==>
account Assets
	note Assets
	check commodity == "€"

**Comment by gour** [comment]
In my ledger file I have:

`commodity 1,000,000.00 $` and that works.

However I do not use `note` and `check commodity` since hledger does not support them.

**Comment by sdementen** [comment]
so you remove from the output the "check commodity" line ?

you can see a rendered ledger at the bottom of the doc https://piecash.readthedocs.io/en/develop/doc/doc.html#the-piecash-command-line-interface

**Comment by gour** [comment]
> so you remove from the output the "check commodity" line ?

Yes, I just comment them.

**Comment by sdementen** [comment]
@gour I have released 1.1 with the currency format in ledger. 
if you find new issues, feel free to report them on github
